sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -52,326 +51,383 @@ private Transforms() {
-            // adding other detected extra fields
-            extractedReceiptItem.setExtractedFields(extractedFieldMap);
-            extractedReceiptList.add(extractedReceiptItem);
+                extractedFormList.add(
+                    new RecognizedForm(extractedFieldMap, formType.get(),
+                        new PageRange(pageNumber, pageNumber),
+                        new IterableStream<FormPage>(Arrays.asList(formPages.get(pageNumber - 1)))));
+            }
         }
+        return extractedFormList;
+    }
+
+    /**
+     * Helper method to transform the service returned {@link AnalyzeResult} to SDK model {@link RecognizedReceipt}.
+     *
+     * @param analyzeResult The service returned result for analyze receipts.
+     * @param includeTextDetails Boolean to indicate if to set reference elements data on fields.
+     *
+     * @return The IterableStream of {@code RecognizedReceipt}.
+     */
+    static IterableStream<RecognizedReceipt> toReceipt(AnalyzeResult analyzeResult, boolean includeTextDetails) {
+        List<RecognizedReceipt> extractedReceiptList =
+            toRecognizedForm(analyzeResult, includeTextDetails).stream()
+                .map(recognizedForm ->
+                    new RecognizedReceipt("en-US", recognizedForm)).collect(Collectors.toList());
+
         return new IterableStream<>(extractedReceiptList);
     }
 
     /**
-     * Helper method that converts the incoming service field value to one of the strongly typed SDK level {@link FieldValue} with
-     * reference elements set when {@code includeTextDetails} is set to true.
+     * Helper method to transform the service returned {@link AnalyzeResult} to SDK model {@link FormPage}.
      *
+     * @param analyzeResult The service returned result for analyze layouts.
+     *
+     * @return The IterableStream of {@code FormPage}.
+     */
+    static List<FormPage> toRecognizedLayout(AnalyzeResult analyzeResult) {
+        List<ReadResult> readResults = analyzeResult.getReadResults();
+        List<PageResult> pageResults = analyzeResult.getPageResults();
+        List<FormPage> formPages = new ArrayList<>();
+
+        for (int i = 0; i < readResults.size(); i++) {
+            ReadResult readResultItem = readResults.get(i);
+            PageResult pageResultItem;
+            List<FormTable> perPageTableList = new ArrayList<>();
+
+            if (!CoreUtils.isNullOrEmpty(pageResults)) {
+                pageResultItem = pageResults.get(i);
+                perPageTableList = getPageTables(pageResultItem, pageResultItem.getPage());
+            }
+
+            // add form lines
+            List<FormLine> perPageFormLineList = new ArrayList<>();
+            if (!CoreUtils.isNullOrEmpty(readResultItem.getLines())) {
+                perPageFormLineList = getReadResultFormLines(readResultItem);
+            }
+
+            // get form tables
+            formPages.add(getFormPage(readResultItem, perPageTableList, perPageFormLineList));
```

Comment:
I've been seeing a lot of locations where an empty list will be passed into a constructor, what happens if `null` is passed? Is this to be a convenience for consumers of the library? If not it appears there will be many object pointers to empty lists.

Reply:
I have updated the places where an empty list was created beforehand and have initialized it to null. It is acceptable for user scenarios to get back null rather than empty lists.